File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = collections.defaultdict()
        for num in nums:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1
        sorted_count = sorted(count.items(), key=lambda x: x[1], reverse=True)
        return [key for key, val in sorted_count[:k]]

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;
        for (int num: nums) {
            count[num]++;
        }

        vector<pair<int, int>> sortedC;
        for (const auto& p: count) {
            sortedC.push_back({p.second, p.first});  // {frequency, number}
        }
        sort(sortedC.rbegin(), sortedC.rend());

        vector<int> res;
        for (int i = 0; i < k; i++) {
            res.push_back(sortedC[i].second);
        }
        return res;
    }
};